BitbucketServerAutomation/Functions/Invoke-BBServerRestMethod.ps1

```diff
@@ -68,6 +72,21 @@ function Invoke-BBServerRestMethod
     Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
 
     $uriPath = 'rest/{0}/{1}/{2}' -f $ApiName.Trim('/'),$Connection.ApiVersion.Trim('/'),$ResourcePath.Trim('/')
+
+    if ($Parameter)
+    {
+        $requestQueryParameter = ''
+        foreach ($key in $Parameter.Keys)
+        {
+            $requestQueryParameter += '&{0}={1}' -f $key, $Parameter[$key]
+        }
+
+        $requestQueryParameter = $requestQueryParameter.TrimStart('&')
+        $requestQueryParameter = [Uri]::EscapeUriString($requestQueryParameter)

```

Comment:
`EscapeUriString` doesn't escape `&` or `=` characters, which would let users inject arbitrary parameters. Instead, use `EscapeDataString` on the query string names and values:

```
$requestQueryParameter += '&{0}={1}' -f [uri]::EscapeDataString($key), [uri]::EscapeDataString($Parameter[$key])
```

Reply:
Oh good idea, I'll make that change.